### PR TITLE
Userspace live patching

### PR DESCRIPTION
**Pull request by dmpop**
### PR creator: Description

Add the "Userspace live patching" chapter

### PR creator: Are there any relevant issues/feature requests?

* jsc#SLE-20214

### PR creator: Which product versions do the changes apply to?

When opening a PR, check all versions of the documentation that your PR applies to.

- SLE 15/openSUSE Leap 15.x
  - [ ] SLE 15 SP4/openSUSE Leap 15.4 *(current `main`, no backport necessary)*
  - [ ] SLE 15 SP3/openSUSE Leap 15.3
  - [ ] SLE 15 SP2/openSUSE Leap 15.2
  - [ ] SLE 15 SP1
  - [ ] SLE 15 GA
- SLE 12
  - [ ] SLE 12 SP5
  - [ ] SLE 12 SP4
  - [ ] SLE 12 SP3

### PR reviewer only: Have all backports been applied?

The doc team member merging your PR will take care of backporting to older documents.
When opening a PR, do *not* set the following check box.

- [ ] all necessary backports are done

### CODE REVIEW

**Review by giulianobelinassi — APPROVED** [review]
Everything is okay for now. As for that dangling script, it will be removed when it be merged into the `ulp` tool.

File: xml/ulp.xml [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    </para>
+   </sect3>
+   <sect3 xml:id="sec-ulp-livepatch-check">
+    <title>Checking if a library is live-patchable</title>
+    <para>
+     To check whether a library is live-patchable, use the following script:
+    </para>
+<screen>#!/bin/sh
+
+is_lib_livepatcheable() {
```
Comment (dariavladykina) [inline xml/ulp.xml:104]:
```suggestion
is_lib_livepatchable() {
```

File: xml/ulp.xml [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+  fi
+
+  local ulp_prologue_dec=`expr $addr_decimal - $ULP_PRE_NOPS_LEN`
+  local ulp_prologue_addr=`printf "0x%lx" $ulp_prologue_dec`
+
+  # Get bytes at target library. It should have $ULP_NOPS_LEN nops (0x90 on x86)
+  local insns=`xxd -s $ulp_prologue_addr -l $ULP_PRE_NOPS_LEN -c $ULP_PRE_NOPS_LEN \
+    -g $ULP_PRE_NOPS_LEN $lib_path | grep -oEi "($ULP_NOP_OPCODE){$ULP_PRE_NOPS_LEN}"`
+
+  # If library is not livepatcheable, insns is empty.
```
Comment (dariavladykina) [inline xml/ulp.xml:131]:
```suggestion
  # If library is not live-patchable, insns is empty.
```

File: xml/ulp.xml [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+  return 0
+}
+
+if [ -z $1 ]; then
+  echo "usage: $0 &lt;.so_to_check&gt;"
+  exit 1
+fi
+
+is_lib_livepatcheable $1
```
Comment (dariavladykina) [inline xml/ulp.xml:144]:
```suggestion
is_lib_livepatchable $1
```

**Review by dariavladykina — COMMENTED** [review]
Hi Dima,

I'm not sure there is a need to correct the spelling in 'is_lib_livepatcheable $1' as it looks like a command so I have suggested it just to find out

**Pull request merged**